ui/src/message/Messages.tsx

```diff
@@ -1,26 +1,118 @@
+import {useAppDispatch, useAppSelector} from '../store';
+import {getAppName} from '../store/app-actions.ts';
+import {fetchMessages, removeMessagesByApp, removeSingleMessage} from '../store/message-actions.ts';
 import Message from './Message';
-import {observer} from 'mobx-react';
-import {inject, Stores} from '../inject';
-import {observable} from 'mobx';
-import ReactInfinite from 'react-infinite';
-import {IMessage} from '../types';
 import ConfirmDialog from '../common/ConfirmDialog';
 import LoadingSpinner from '../common/LoadingSpinner';
 
-type IProps = RouteComponentProps<{id: string}>;
+const Messages = () => {
+    const dispatch = useAppDispatch();
+    const {id} = useParams();
+    const appId = id !== undefined ? parseInt(id as string) : -1;
 
-interface IState {
-    appId: number;
-}
+    const [toDeleteAll, setToDeleteAll] = useState<boolean>(false);
+
+    const heights: Record<string, number> = {};
+
+    const selectedApp = useAppSelector((state) => state.app.items.find((app) => app.id === appId));
+    const apps = useAppSelector((state) => state.app.items);
+    const messages = useAppSelector((state) => appId === -1 ? state.message.items : state.message.items.filter((item) => item.appid === appId));
+    const hasMore = useAppSelector((state) => state.message.hasMore);
+    const name = dispatch(getAppName(appId));
+    const messagesLoaded = useAppSelector((state) => state.message.loaded);
+    const hasMessages = messages.length !== 0;
 
+    useEffect(() => {
+        dispatch(fetchMessages(appId));
+    }, [dispatch, appId]);
+
+    const label = (text: string) => (
+        <Grid size={12}>
+            <Typography variant="caption" component="div" gutterBottom align="center">
+                {text}
+            </Typography>
+        </Grid>
+    );
+
+    return (
+        <DefaultPage
+            title={name}
+            rightControl={
+                <div>
+                    <Button
+                        id="refresh-all"
+                        variant="contained"
+                        color="primary"
+                        onClick={() => dispatch(fetchMessages(appId))}
+                        style={{marginRight: 5}}>
+                        Refresh
+                    </Button>
+                    <Button
+                        id="delete-all"
+                        variant="contained"
+                        disabled={!hasMessages}
+                        color="primary"
+                        onClick={() => setToDeleteAll(true)}>
+                        Delete All
+                    </Button>
+                </div>
+            }>
+            {!messagesLoaded ? (
+                <LoadingSpinner />
+            ) : hasMessages ? (
+                <div style={{width: '100%'}} id="messages">
+                    {/* TODO: maybe replace ReactInfitite with react-window, which is also documented here: https://mui.com/material-ui/react-list/#virtualized-list */}
```

Comment:
I've recently tried a lot of window libraries for another project and landed on https://tanstack.com/virtual/latest/docs/introduction react-window doesn't play to well with having a window over the whole screen.